neqo-http3/src/hframe.rs

```diff
@@ -104,23 +207,7 @@ impl HFrame {
             }
             HFrame::Settings { settings } => {
                 enc.encode_vvec_with(|enc_inner| {
-                    for iter in settings.iter() {
-                        match iter.0 {
-                            HSettingType::MaxHeaderListSize => {
-                                enc_inner.encode_varint(SETTINGS_MAX_HEADER_LIST_SIZE as u64);
-                                enc_inner.encode_varint(iter.1);
-                            }
-                            HSettingType::MaxTableSize => {
-                                enc_inner.encode_varint(SETTINGS_QPACK_MAX_TABLE_CAPACITY as u64);
-                                enc_inner.encode_varint(iter.1);
-                            }
-                            HSettingType::BlockedStreams => {
-                                enc_inner.encode_varint(SETTINGS_QPACK_BLOCKED_STREAMS as u64);
-                                enc_inner.encode_varint(iter.1);
-                            }
-                            HSettingType::UnknownType => {}
-                        }
-                    }
+                    settings.encode_settings(enc_inner);
```

Comment:
Maybe move the outer `encode_vvec_with` and rename the function to `encode_frame` or `encode_frame_contents` (and change `decode_settings` to `decode_frame` to match)